Yosemite/Yosemite/Model/Mocks/ActionHandlers/MockStatsActionV4Handler.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+import Storage
+import Networking
+
+struct MockStatsActionV4Handler: MockActionHandler {
+    typealias ActionType = StatsActionV4
+
+    let objectGraph: MockObjectGraph
+    let storageManager: StorageManagerType
+
+    func handle(action: ActionType) {
+        switch action {
+            case .retrieveStats(let siteID, let timeRange, _, _, _, let onCompletion):
+                retrieveStats(siteID: siteID, timeRange: timeRange, onCompletion: onCompletion)
+            case .retrieveSiteVisitStats(let siteID, _, let timeRange, _, let onCompletion):
+                retrieveSiteVisitStats(siteID: siteID, timeRange: timeRange, onCompletion: onCompletion)
+            case .retrieveTopEarnerStats(let siteID, let timeRange, _, _, let onCompletion):
+                retrieveTopEarnerStats(siteID: siteID, timeRange: timeRange, onCompletion: onCompletion)
+            default: unimplementedAction(action: action)
+        }
+    }
+
+    func retrieveStats(siteID: Int64, timeRange: StatsTimeRangeV4, onCompletion: @escaping (Error?) -> ()) {
+        let store = StatsStoreV4(dispatcher: Dispatcher(), storageManager: storageManager, network: NullNetwork())
+
+        switch timeRange {
+            case .today: success(onCompletion)
+            case .thisWeek: success(onCompletion)
+            case .thisMonth: success(onCompletion)
+            case .thisYear:
+                store.upsertStoredOrderStats(readOnlyStats: objectGraph.thisYearOrderStats, timeRange: timeRange)
+                onCompletion(nil)
```

Comment:
My OCD speaking again: feels a bit unbalanced and harder to read to have some `case` having their action on the same line but others (just the last one here) being on separate lines (because there's more than one expression).

At first skim my mind read it as if that was a C-style switch (i.e. with implicit `fallthrough`) and thought that all 4 cases were executing the last 2 lines (`store.upsert…` + `onCompletion`), before I clicked back into realizing there were some `success(onCompleteion)` calls hiding there.

It might only be a nit but even if it would make the code spread a bit more vertically, I think it would be worth always putting the instructions after each case on their own lines. In this specific example, given the instructions are the same for the 3 first cases, we could DRY the 3 cases on one line though:

```suggestion
            case .today, .thisWeek, .thisMonth:
                success(onCompletion)
            case .thisYear:
                store.upsertStoredOrderStats(readOnlyStats: objectGraph.thisYearOrderStats, timeRange: timeRange)
                onCompletion(nil)
```

Ditto for all other similar `switch` in the same file below ofc.

Reply:
Yeah fair enough – fixed all of the cases in this file in 11e4afa0a6e8ae757a51e5a501f8f4b2b50a8886 and verified that there aren't others in the other files.